rustls/src/msgs/message.rs

```diff
@@ -91,23 +92,23 @@ impl OpaqueMessage {
         //  implemented per section 5.1 of RFC8446 (TLSv1.3)
         //              per section 6.2.1 of RFC5246 (TLSv1.2)
         if typ != ContentType::ApplicationData && len == 0 {
-            return Err(MessageError::IllegalLength);
+            return Err(MessageError::EmptyDataForDisallowedRecord);
```

Comment:
This variant name doesn't really make sense to me, what are you trying to convey here? It seems to be much more abstract than the `EmptyApplicationData` I proposed.

Reply:
I'm fine working on the errors name more. I had changed it because, AFAICT, its not actually empty application data that is being checked for here. According to the RFC section (and the PR that added this, https://github.com/rustls/rustls/pull/848), application data is the only content type that's allowed to be empty, but not the others. Given that, I tried to come up with a name inverse to `EmptyApplicationData`.

Reply:
Ah yes, sorry, I missed that. How about `InvalidEmptyPayload`?

Reply:
That works for me. I think its more clear when combined with the doc comments then what I had.